tests/e2e/test_main.py

```diff
@@ -237,13 +244,62 @@ def test_image_build_custom_build_args(cli_runner: CLIRunner) -> None:
     assert f"arg-another-{tag}" in result.stdout
 
 
+@pytest.mark.skipif(sys.platform == "win32", reason="kaniko does not work on Windows")
+def test_image_build_env(cli_runner: CLIRunner) -> None:
+    result = cli_runner(["neuro-extras", "init-aliases"])
+    assert result.returncode == 0, result
+
+    result = cli_runner(
+        ["neuro", "secret", "add", "gh-token-e2e", "$TEST_GITHUB_TOKEN"]
+    )
+    assert result.returncode == 0, result
+
+    dockerfile_path = Path("nested/custom.Dockerfile")
+    dockerfile_path.parent.mkdir(parents=True)
+
+    github_url = (
+        "https://raw.githubusercontent.com/neuromation/platform-api/master/setup.py"
+    )
+
+    with open(dockerfile_path, "w") as f:
+        f.write(
+            textwrap.dedent(
+                f"""\
+                FROM ubuntu:latest
+                ARG GIT_TOKEN
```

Comment:
I thought `ARG` is for `docker build --build-arg`?

Reply:
Right. We use `ARG` / `--build-arg` combination to pass secret values to the image being built.